src/PayID/pay-id-client.ts

```diff
@@ -22,7 +24,45 @@ export default class PayIDClient {
    * @param payID The payID to resolve for an address.
    * @returns An XRP address representing the given PayID if one exists, otherwise undefined.
    */
-  public async xrpAddressForPayID(_payID: string): Promise<string | undefined> {
-    throw PayIDError.unimplemented
+  public async xrpAddressForPayID(payID: string): Promise<string | undefined> {
+    const paymentPointer = PayIDUtils.parsePaymentPointer(payID)
+    if (!paymentPointer) {
+      throw PayIDError.invalidPaymentPointer
+    }
+
+    const url = `https://${paymentPointer.host}${paymentPointer.path}`
+    // TODO(keefertaylor): Attach network parameter.
+    // TODO(keefertaylor): Generalize the below to allow requesting in other types.
+    const requestConfig = {
+      headers: {
+        Accept: 'application/xrp+json',
+      },
+    }
+
+    return this.axiosInstance
+      .get(url, requestConfig)
+      .then((result: AxiosResponse) => {
+        // TODO(keefertaylor): properly extract xrp address and tag, format as X-Address.
```

Comment:
I think we may try to "standardize" the XRP format for PayID to be something like:
```
{
  "xAddress": "XVdQPCEqWheLp7JNnWE3HbenVZof8jfEHcbWd7Vr9MVXPWb",
  "classicAddress": "rMFmeAsBdfiQFcdFszwPmsj4GT82EFfLHe",
  "destinationTag": "12345"
}
```

In that case, you wouldn't need to parse the address and tag (although it might still be worth handling that case).

Reply:
Although, `doug.purdy.im` just returns `address: r58...`, so this is in flux too, and if you want to prioritize working with `doug.purdy.im`, that's fine.

https://doug.purdy.im/.well-known/pay